real-main/app/models/user/manager.py

```diff
@@ -504,27 +504,48 @@ def on_user_delete_delete_cognito(self, user_id, old_item):
             # TODO: catch 404 error & log warning
             self.cognito_client.delete_identity_pool_entry(user_id)
 
-    def find_users(self, caller_user, emails=None, phones=None):
+    def find_contacts(self, caller_user, contacts):
         """
         Given a list of emails and a list of phones, return a list of user_ids of users
         in our system with those emails and phones.
         For each returned user_id that is not already following the user that called this
         method, create a card inviting them to follow.
         """
-        emails = emails or []
-        phones = phones or []
 
-        user_ids_from_emails = self.email_dynamo.batch_get_user_ids(emails)
-        user_ids_from_phones = self.phone_number_dynamo.batch_get_user_ids(phones)
-        user_ids = list(set(user_ids_from_emails + user_ids_from_phones))  # uniquify
+        found_contacts = []
+        email_contacts, phone_contacts = {}, {}
+        for contact in contacts:
+            for email in contact.get('emails', []):
+                email_contacts[email] = contact['contactId']
```

Comment:
```suggestion
            email_contacts.update({email: contact['contactId'] for email in contact.get('emails, [])})
```

Same thing can be done for phone, of course.